src/assets/src/components/queueSettings.tsx

```diff
@@ -0,0 +1,315 @@
+        setName(newName);
+        validateAndSetStringResult(newName, queueNameSchema, setNameValidationResult, true);
+        if (showSuccessMessage) setShowSuccessMessage(false);
+    };
+    const handleDescriptionChange = (newDescription: string) => {
+        setDescription(newDescription);
+        validateAndSetStringResult(newDescription, queueDescriptSchema, setDescriptValidationResult, true);
+        if (showSuccessMessage) setShowSuccessMessage(false);
+    };
+    const handleAllowedChange = (newAllowedBackends: Set<string>) => {
+        setAllowedMeetingTypes(newAllowedBackends);
+        validateAndSetMeetingTypesResult(newAllowedBackends, setAllowedValidationResult, queue)
+        if (showSuccessMessage) setShowSuccessMessage(false);
+    };
+
+    // On click handlers
+    const handleSaveGeneralClick = () => {
+        const curNameValidationResult = !nameValidationResult
+            ? validateAndSetStringResult(name, queueNameSchema, setNameValidationResult, true)
+            : nameValidationResult;
+        const curDescriptValidationResult = !descriptValidationResult
+            ? validateAndSetStringResult(description, queueDescriptSchema, setDescriptValidationResult, true)
+            : descriptValidationResult;
+        const curAllowedValidationResult = !allowedValidationResult
+            ? validateAndSetMeetingTypesResult(allowedMeetingTypes, setAllowedValidationResult, queue)
+            : allowedValidationResult;
+
+        if (!curNameValidationResult!.isInvalid && !curDescriptValidationResult!.isInvalid && !curAllowedValidationResult!.isInvalid) {
+            const nameForUpdate = name.trim() !== queue?.name ? name : undefined;
+            const descriptForUpdate = description.trim() !== queue?.description ? description : undefined;
+            const allowedForUpdate = !compareStringArrays(Array.from(allowedMeetingTypes), queue!.allowed_backends)
+                ? allowedMeetingTypes : undefined;
+            if (nameForUpdate || descriptForUpdate || allowedForUpdate) {
+                doUpdateQueue(nameForUpdate, descriptForUpdate, allowedForUpdate);
+                setShowSuccessMessage(true);
+                resetValidationResults();
+            }
+        } else {
+            if (!showCorrectGeneralMessage) setShowCorrectGeneralMessage(true);
+        }
+    };
+    const handleDiscardGeneralClick = () => {
+        setName(queue!.name);
+        setDescription(queue!.description);
+        setAllowedMeetingTypes(new Set(queue!.allowed_backends));
+        resetValidationResults();
+    }
+
+    const handleTabSelect = (eventKey: string) => {
+        if (Object.values(AvailableTabs).some(value => value === eventKey)) {
+            const typedKeyString = eventKey as keyof typeof AvailableTabs;
+            setActiveKey(AvailableTabs[typedKeyString]);
+        } else {
+            throw Error(`eventKey "${eventKey}" is not defined in enum AvailableTabs`);
+        }
+    };
```

Comment:
I think you mentioned this before, but maybe we can just pare this down to:
```
const handleTabSelect = (eventKey: string) => {
    setActiveKey(eventKey as AvailableTabs);
};
```
if there's no serious risk of using a string that isn't present in the enum.

Reply:
I think this is a similar but different case, but yeah, is this this what you were thinking? https://github.com/tl-its-umich-edu/remote-office-hours-queue/pull/173/commits/ef09509e5687a3c0548cf45513c20129c81745cd

I switched the `activeKey` variable to use an `enum` member, since I was getting some weird errors when trying to use a `type` declaration. Thus, I also need to use `typeof keyof` (which I came across in my research).

Reply:
I started looking at this more carefully, and it's not actually working like I think it is. `AvailableTabs[eventKey as keyof typeof AvailableTabs]` seems to return `undefined`. Hm...

Reply:
Seems like reverse mapping of string values back to enum member keys doesn't work. People are recommending a https://stackoverflow.com/questions/54039138/how-to-handle-reverse-mapping-of-string-enums-with-different-string-literals-in

The thing is, the tabs still mostly work, except on the first click. I tried removing all the tab management code entirely, and it actually works fine without it. So I'm going to go in that direction.

Reply:
Yeah, this is just simpler all around, to let the default `Tab.Container` react-boostrap component handle the Tab changes in `queueSettings`.

I think it's still appropriate to override `onSelect` and `activeKey` in the `Tab.Container` in `addQueue.tsx`, that's a workflow and we need to prevent folks from proceeding if the first page is invalid.